src/App.tsx

```diff
@@ -15,48 +16,96 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
+export enum SortOptions {
+  Alphabetically = 'alphabetically',
+  Length = 'length',
+  Reset = 'reset',
+}
+
+const handleSort
+  = (goods: string[],
+    options: { activeSort: SortOptions; isReversed: boolean }) => {
+    const preparedGoods = [...goods];
+
+    preparedGoods.sort((a, b) => {
+      switch (options.activeSort) {
+        case SortOptions.Alphabetically:
+          return a.localeCompare(b);
+
+        case SortOptions.Length:
+          return a.length - b.length;
+
+        default:
+          return 0;
+      }
+    });
+
+    if (options.isReversed) {
+      preparedGoods.reverse();
+    }
+
+    return preparedGoods;
+  };
+
 export const App: React.FC = () => {
+  const [activeSort, setActiveSort]
+    = useState('');
+
+  const [isReversed, setIsReversed]
+    = useState(false);
+
+  const visibleGoods = handleSort(goodsFromServer, {
+    activeSort: activeSort || SortOptions.Alphabetically,
+    isReversed,
+  } as { activeSort: SortOptions; isReversed: boolean });
+
   return (
     <div className="section content">
       <div className="buttons">
         <button
           type="button"
-          className="button is-info is-light"
+          className={cn('button', 'is-info',
+            { 'is-light': activeSort !== 'alphabetically' })}
+          onClick={() => setActiveSort('alphabetically')}
         >
           Sort alphabetically
         </button>
 
         <button
           type="button"
-          className="button is-success is-light"
+          className={cn('button', 'is-success',
+            { 'is-light': activeSort !== 'length' })}
+          onClick={() => setActiveSort('length')}
         >
           Sort by length
         </button>
 
         <button
           type="button"
-          className="button is-warning is-light"
+          className={cn('button', 'is-warning', { 'is-light': !isReversed })}
+          onClick={() => setIsReversed(!isReversed)}
```

Comment:
The same
```suggestion
          onClick={reverse}
```

Reply:
Not fixed

Reply:
<img width="590" alt="image" src="https://github.com/mate-academy/react_list-of-goods-hooks/assets/70220667/8c95f35c-e52c-41dd-b181-8a0e68e55f65">